docs/library/stm.rst

```diff
@@ -102,3 +102,11 @@ the second CPU, the RF core.
     Execute a HCI command on the SYS channel.  The execution is synchronous.
 
     Returns a bytes object with the result of the SYS command.
+
+.. function:: rfcore_ble_hci(ogf, ocf, data) or rfcore_ble_hci(buffer)
```

Comment:
Since the first form is not used (yet!) I'd say just implement the second form.  It can be enhanced later if anyone ever needs it (BLE HCI commands are arguably different to SYS HCI commands, so I don't think it's a problem having a different function signature here, compared to `rfcore_sys_hci`).

Also, it could be: `rfcore_ble_hci(cmd, outbuf) -> int` where `outbuf` is a bytearray used to store the result, and it returns the number of bytes copied into that buffer.  Then this function would be fully in-place (no heap allocation needed).  Would you consider implementing that instead?  I don't think it would change much (`_stm32wb55_transparent.c` already has a big 1024 byte buffer which could be reused for the `outbuf`).

Reply:
Thanks @dpgeorge I've removed that second / unused form of the function, this does simplify things.

I've also added the second response_buf arg as optional - it certainly is a worthwhile option to allow no-alloc useage of the function. If the buffer arg isn't supplied I've kept it returing the fully-formed bytes object as this simpler usage could be handy for simple use cases.

I've used this buffer arg for the native module too as suggested, it's working well and cleaner overall.